### PR TITLE
add googlenet into image classifier example

### PR DESCRIPTION
**Pull request by tigercosmos**
## Description

This PR adds the googlenet into `example/image_classifier`

## Type of change

add an example

## Checklist:

- [X] Have you added tests that prove your fix is effective or that this feature works?
- [X] New and existing unit tests pass locally with these changes?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [X] Have you made corresponding changes to the documentation?

### CODE REVIEW

**Comment by msaroufim** [comment]
Thank you for your contribution. I think your PR made me realize that it may be best to refactor one of the existing examples to make it easier to support new models. Let me know if you'd like to take a stab at it because 50 lines of documentation for 3 lines of code for each new model doesn't make the most sense

**Comment by tigercosmos** [comment]
@msaroufim I strongly agree with your point. However, I am afraid that I don't have time to put more effort into this refactoring.

**Comment by msaroufim** [comment]
That's fair, I opened up another issue to track the code duplication issue #1047 so I think we can pull this example in. Thank you for your contribution to Pytorch!

**Review by msaroufim — APPROVED** [review]

File: examples/image_classifier/googlenet/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+mv googlenet.mar model_store/
+torchserve --start --model-store model_store --models googlenet
+curl http://127.0.0.1:8080/predictions/googlenet -T ./serve/examples/image_classifier/kitten.jpg
+```
+
+#### TorchScript example using googlenet image classifier:
+
+* Save the alexnet model in as an executable script module or a traced script:
+
+1. Save model using scripting
```
Comment (HamidShojanazeri) [inline examples/image_classifier/googlenet/README.md:19]: @tigercosmos running the torchscript version, facing this [error](https://gist.github.com/HamidShojanazeri/f49b3f79a981d5b89bcf3cbac33c6f65) on post-processing. Wonder if you can repro, on your end.

File: examples/image_classifier/googlenet/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+   ```python
+   #scripted mode
+   from torchvision import models
+   import torch
+   model = models.googlenet(pretrained=True)
+   sm = torch.jit.script(model)
+   sm.save("googlenet.pt")
+   ```
+
+2. Save model using tracing
```
Comment (HamidShojanazeri) [inline examples/image_classifier/googlenet/README.md:29]: @tigercosmos I haven't been able to trace the mode, failing with assertion [error](https://gist.github.com/HamidShojanazeri/9ac3cfaad8d214770090edbd97b24c31) in trace.

File: examples/image_classifier/googlenet/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+   traced_script_module.save("googlenet.pt")
+   ```  
+ 
+* Use following commands to register googlenet torchscript model on TorchServe and run image prediction
+
+    ```bash
+    torch-model-archiver --model-name googlenet --version 1.0  --serialized-file googlenet.pt --extra-files ./serve/examples/image_classifier/index_to_name.json --handler image_classifier
+    mkdir model_store
+    mv googlenet.mar model_store/
+    torchserve --start --model-store model_store --models googlenet
```
Comment (HamidShojanazeri) [inline examples/image_classifier/googlenet/README.md:47]: @tigercosmos Please change to --models googlenet = googlenet.mar

**Review by HamidShojanazeri — CHANGES_REQUESTED** [review]
Thanks @tigercosmos for your contribution, please refer to the inline comments.

**Pull request closed without merging**